app/upload/UploadImage.tsx

```diff
@@ -0,0 +1,144 @@
+"use client";
```

Comment:
Tady dochází k tomu varování ohledně serializovatelných props. Nerozumím tomu stopro, ale mám nějakou představu :)

Stručně řečeno: Ta direktiva `use client` není úplně šťastně nazvaná, protože působí, jako bys takhle měla označit _každou klientskou komponentu_. Ve skutečnosti ale označuje _rozhraní mezi klientem a serverem_. Což tady není, protože už ta nadřazená komponenta je klientská. Proto můžeš tu direktivu tady bezpečně smazat. Doufám :)

(Proto ti to taky reálně fungovalo i přes varování – komponenta `UploadImage` je vložená do jiné klientské komponenty, takže i když některé ty její props nejsou serializovatelné, nevadí to, protože se předávají jen mezi dvěma komponentami na klientovi a tím pádem se nemusí serializovat pro přenos po síti.)

Reply:
aha, cca rozumiem :D cize je to preto ok ze ked rodicovske komponenty su klientske, tym padom uz nie je vyzadovane priamo potomkov oznacit `use client`? Je to vlastne preto ze `SignedInPage` je klientska a tym padom je komunikacia medzi `UserProfileTab` a `UploadImage` v poriadku aj skrze predavanie funkcii?

Anyway, zmazala som to a vyzera ze to ide a chyba zmizla cize super :D

Reply:
Jo, přesně tak tomu rozumím.